File: DailyFlash Solutions/DailyFlash Solutions/33-DailyFlash_Solutions/20_Feb_Solution_Three/Python/prog5.py
num = int(input("num : "))
k = 0
temp = num
while(num != 0):
    rem = num % 10
    s = 0
    for i in range(1, int(rem / 2) + 1):
        if(rem % i == 0):
            s += i
    if(s == rem):
        print(s, end = " ")
        k += 1
    num = int(num / 10)
if(k == 0):
    print("The number", temp, "does not have any digit, which is perfect in nature.")

File: DailyFlash Solutions/DailyFlash Solutions/33-DailyFlash_Solutions/20_Feb_Solution_Three/C/prog5.c
#include<stdio.h>

int main(void) {
	
	int num, rem, flag = 0, sum;
	printf("Enter num : ");
	scanf("%d", &num);
	int temp = num;
	while(num != 0) {
		rem = num % 10;
		sum = 0;
		for(int i = 1; i <= rem / 2; i++) {
			if(rem % i == 0) 
				sum += i;
		}
		if(sum == rem) {
			printf("%d ", rem);
			flag++;
		}
		num /= 10;
	}
	if(flag == 0)
		printf("The num %d not have any digit ehich is perfect in nature.", temp);
	return 0;
}
